plugins/utils/toggleterm.nix

```
@@ -0,0 +1,160 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.toggleterm;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.toggleterm = {
+    enable = mkEnableOption "toggleterm";
+
+    package = helpers.mkPackageOption "toggleterm" pkgs.vimPlugins.toggleterm-nvim;
+
+    size =
+      helpers.defaultNullOpts.mkNullable
+      (with types; either number str) "" ''
+        size can be a number or function which is passed the current terminal
+        Example:
+        size = 20
+        OR
+        size = function(term)
+          if term.direction == "horizontal" then
+            return 15
+          elseif term.direction == "vertical" then
+            return vim.o.columns * 0.4
+          end
+        end
+      '';
```

```suggestion
    size =
      helpers.defaultNullOpts.mkNullable
      (with types; either number str) "" ''
        `size` can be a number or function which is passed the current terminal.
        Example:
        ```
          size = 20
        ```
        OR
        ```
          size = function(term)
            if term.direction == "horizontal" then
              return 15
            elseif term.direction == "vertical" then
              return vim.o.columns * 0.4
            end
          end
        ```
      '';
```

Try to wrap code with "``" in documentation.